api/pkg/api/main.go

```diff
@@ -127,7 +129,24 @@ func saveHarToDb(entry *har.Entry, connectionInfo *tap.ConnectionInfo) {
 	)
 	if k8sResolver != nil {
 		resolvedSource = k8sResolver.Resolve(connectionInfo.ClientIP)
-		resolvedDestination = k8sResolver.Resolve(fmt.Sprintf("%s:%s", connectionInfo.ServerIP, connectionInfo.ServerPort))
+		if resolvedSource == "" {
+			rlog.Debug("Cannot find resolved name to source: %s\n", connectionInfo.ClientIP)
+			if os.Getenv("SKIP_NOT_RESOLVED_SOURCE") == "1" {
+				return
+			}
+		}
+		unresolvedDestination := fmt.Sprintf("%s:%s", connectionInfo.ServerIP, connectionInfo.ServerPort)
+		resolvedDestination = k8sResolver.Resolve(unresolvedDestination)
+		if resolvedDestination == "" {
+			rlog.Debug("Cannot find resolved name to dest: %s\n", unresolvedDestination)
+			if os.Getenv("SKIP_NOT_RESOLVED_DEST") == "1" {
+				return
+			}
+		}
+
+		if resolvedDestination == "" || resolvedSource == "" {
+			rlog.Debug("BUG:: %s-%s (after resolving: %s -> %s)\n", connectionInfo.ClientIP, unresolvedDestination, resolvedSource, resolvedDestination)
```

Comment:
Save `connectionInfo.ClientIP` to `unresolvedSource` and print it here.
Otherwise this debug message will "break" when line 131 changes.